controller/transfer/util.py

```diff
@@ -98,7 +99,14 @@ def convert_to_record_dict(
         os.remove(file_name)
     run_limit_checks(df, project_id, user_id)
     run_checks(df, project_id, user_id)
-    return df.to_dict(orient="records")
+    check_and_convert_category_for_unknown(df)
+    return df.to_dict("records")
+
+
+def check_and_convert_category_for_unknown(df_check: pd.DataFrame) -> None:
+    for key in df_check.columns:
+        if category.infer_category_enum(df_check, key) == enums.DataTypes.UNKNOWN.value:
+            df_check[key] = df_check[key].astype(str)
```

Comment:
if something was changed we should notify the user
```suggestion
    changed_keys = []
    for key in df_check.columns:
        if category.infer_category_enum(df_check, key) == enums.DataTypes.UNKNOWN.value:
            changed_key.append(key)
            df_check[key] = df_check[key].astype(str)
    if len(changed_key) > 0:
        # send notification with args argument ",".join(changed_key)
        # message: Some attribute datatypes(@@arg@@) were converted to string since they are not nativly supported by refinery" 
```